server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -162,7 +163,9 @@ public BitSetProducer bitsetFilter(Query filter) {
     }
 
     public <IFD extends IndexFieldData<?>> IFD getForField(MappedFieldType fieldType) {
-        return (IFD) indexFieldDataService.apply(fieldType, fullyQualifiedIndexName);
+        Objects.requireNonNull(fieldType, "fieldType is null");
+
+        return (IFD) indexFieldDataService.apply(fieldType.fieldTypeForIndex(), fullyQualifiedIndexName);
```

Comment:
if a non alias field type will simply return its name, for aliases it will return the name of the alias target, and the rest will just work!